modules/oke/cloudinit.tf

```diff
@@ -0,0 +1,29 @@
+locals {
+  worker_script_template = templatefile("${path.module}/cloudinit/worker.template.sh",
+    {
+      worker_timezone = "Australia/Sydney"
+    }
+  )
+  # example for adding more script 
+  # second_script_template = templatefile("${path.module}/cloudinit/second.template.sh",{})
+}
+
+# cloud-init for workers
+data "cloudinit_config" "worker" {
+  gzip          = false
+  base64_encode = true
+
+  part {
+    filename     = "worker.sh"
+    content_type = "text/x-shellscript"
+    content      = local.worker_script_template
+  }
+  
+  # example for adding more script
+  # part {
+  #   filename     = "second.sh"
+  #   content_type = "text/x-shellscript"
+  #   content      = local.second_script_template
+  # }
+}
+

```

Comment:
may i know how the cloud init scripts are received from the end user when they are using oke module.They are not supposed to edit any tf files.

Reply:
They have to add their scripts in the cloudinit directory, load them as templates in locals, and then edit the cloudinit_config data source in cloudinit.tf. In this way, it is an explicit action on the user's part to load a customer script whereas the worker.template.sh, they are not supposed to modify.